clouddriver-google/src/main/java/com/netflix/spinnaker/clouddriver/google/deploy/handlers/BasicGoogleDeployHandler.java

```diff
@@ -289,8 +290,45 @@ protected String getMachineTypeNameFromInput(
     if (description.getInstanceType().contains("custom-")) {
       return description.getInstanceType();
     } else {
-      return GCEUtil.queryMachineType(
-          description.getInstanceType(), location, description.getCredentials(), task, BASE_PHASE);
+      List<String> queryZone =
+          description.getRegional()
+              ? (description.getSelectZones()
+                  ? description.getDistributionPolicy().getZones()
```

Comment:
CAN distribution policy be null?  IF SO this would potentially introduce an NPE...

Reply:
Ideally Deck protects us from this since it is failing the validations but i can change to: 
```
      List<String> queryZone =
          description.getRegional()
              ? (description.getSelectZones() && !description.getDistributionPolicy().getZones().isEmpty()
                  ? description.getDistributionPolicy().getZones()
                  : Collections.singletonList(location))
              : Collections.singletonList(location);
```
Which basically is pipeline definition: 
```
      "distributionPolicy": {
        "targetShape": "EVEN",
        "zones": []
      },
      "regional": true,
      "selectZones": true,
```

Reply:
Updated